### PR TITLE
fix(azure): add note about ADU

### LINKED ISSUE #151: Azure: mention that ADU is not supported, yet
https://docs.microsoft.com/en-gb/azure/iot-hub-device-update/
https://github.com/NordicSemiconductor/asset-tracker-cloud-docs/discussions/12#discussioncomment-483383

### PR DESCRIPTION
**Pull request by coderbyheart**
Closes #151

### CODE REVIEW

File: docs/firmware/azure/Index.rst [hunk 1]
```diff
@@ -9,6 +9,12 @@ It is compatible with the :ref:`nRF Asset Tracker for Azure <index_azure>`.
 
 The `firmware GitHub repository <https://github.com/NordicSemiconductor/asset-tracker-cloud-firmware-azure>`_ provides an example for using the application as an out-of-tree development copy, which optionally enables the SUPL client.
 
+.. note::
+
+   The current firmware-over-the-air (FOTA) update mechanism implementated in nRF Asset Tracker for Azure is based on `this tutorial <https://docs.microsoft.com/en-us/azure/iot-hub/tutorial-firmware-update>`_.
+   In February 2021, Microsoft released `a preview of Device Update for IoT Hub <https://techcommunity.microsoft.com/t5/internet-of-things/streamline-device-management-with-device-update-for-iot-hub-now/ba-p/2167663>`_ which will replace the current, as soon as it is generally available.
+   Work on implementing it `has already started <https://github.com/NordicSemiconductor/asset-tracker-cloud-docs/discussions/12#discussioncomment-483383>`_.
```
Comment (peknis) [inline docs/firmware/azure/Index.rst]:
```suggestion
   The current firmware-over-the-air (FOTA) update mechanism implemented in nRF Asset Tracker for Azure is based on `this Azure tutorial <https://docs.microsoft.com/en-us/azure/iot-hub/tutorial-firmware-update>`_.
   In February 2021, Microsoft released `a preview of Device Update for IoT Hub <https://techcommunity.microsoft.com/t5/internet-of-things/streamline-device-management-with-device-update-for-iot-hub-now/ba-p/2167663>`_ which will replace the current mechanism, as soon as it is available in public.
   Work on implementing it `has already started <https://github.com/NordicSemiconductor/asset-tracker-cloud-docs/discussions/12#discussioncomment-483383>`_.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: docs/firmware/azure/Index.rst (hunk 1 above)
Comment (coderbyheart) [reply docs/firmware/azure/Index.rst]: `in public`: the feature is already available in public.
`general availability` [is a term used by vendors like Azure](https://www.google.com/search?q=azure+general+availability&oq=azure+general&aqs=chrome.1.69i57j0l6j46i175i199j0l2.2650j0j7&sourceid=chrome&ie=UTF-8) to denote a feature which is no longer in preview.

File: docs/firmware/azure/Index.rst [hunk 2]
```diff
@@ -9,6 +9,12 @@ It is compatible with the :ref:`nRF Asset Tracker for Azure <index_azure>`.
 
 The `firmware GitHub repository <https://github.com/NordicSemiconductor/asset-tracker-cloud-firmware-azure>`_ provides an example for using the application as an out-of-tree development copy, which optionally enables the SUPL client.
 
+.. note::
+
+   The current firmware-over-the-air (FOTA) update mechanism implementated in nRF Asset Tracker for Azure is based on `this tutorial <https://docs.microsoft.com/en-us/azure/iot-hub/tutorial-firmware-update>`_.
```
Comment (umapraseeda) [inline docs/firmware/azure/Index.rst]:
```suggestion
   The current firmware-over-the-air (FOTA) update mechanism implemented in nRF Asset Tracker for Azure is based on the `Microsoft tutorial on implementing a device firmware update process<https://docs.microsoft.com/en-us/azure/iot-hub/tutorial-firmware-update>`_.
```

File: docs/firmware/azure/Index.rst (hunk 1 above)
Comment (umapraseeda) [inline docs/firmware/azure/Index.rst]:
```suggestion
   See the `Azure support discussion <https://github.com/NordicSemiconductor/asset-tracker-cloud-docs/discussions/12#discussioncomment-483383>`_ for details about the ongoing work to implement the new update mechanism.
```

File: docs/firmware/azure/Index.rst [hunk 3]
```diff
@@ -9,6 +9,12 @@ It is compatible with the :ref:`nRF Asset Tracker for Azure <index_azure>`.
 
 The `firmware GitHub repository <https://github.com/NordicSemiconductor/asset-tracker-cloud-firmware-azure>`_ provides an example for using the application as an out-of-tree development copy, which optionally enables the SUPL client.
 
+.. note::
+
+   The current firmware-over-the-air (FOTA) update mechanism implemented in nRF Asset Tracker for Azure is based on `this Azure tutorial <https://docs.microsoft.com/en-us/azure/iot-hub/tutorial-firmware-update>`_.
```
Comment (umapraseeda) [inline docs/firmware/azure/Index.rst]:
```suggestion
   The current firmware-over-the-air (FOTA) update mechanism implemented in nRF Asset Tracker for Azure is based on the `Microsoft Azure tutorial for implementing a device firmware update process <https://docs.microsoft.com/en-us/azure/iot-hub/tutorial-firmware-update>`_.
```

File: docs/firmware/azure/Index.rst (hunk 2 above)
Comment (coderbyheart) [reply docs/firmware/azure/Index.rst]: First change is fixed, second change conflicts with @peknis suggestion. But I guess I'll take that.

**Review by umapraseeda — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**